### DISCUSSION TITLE
notification trigger.on-rollout-aborted is sending twice, is that normal?

### DISCUSSION THREAD

**Discussion by juanvasquezreyes** [argoproj/argo-rollouts#1964] [General]
I'm using the trigger on-rollout-aborted for notification, but the notification when a rollout is aborted is being sent twice, is that normal? with milliseconds difference

I tried adding [oncePer option ](https://github.com/argoproj/notifications-engine/blob/04ff7456dbfd058d6ca344f827bcb483f40ee924/docs/triggers.md#onceper) by version but still is sending notification twice

I added the phase for tracking purposes and in one notification the rollout phase is "Progressing" the other one is "degraded"

**Comment by nicolasfagotti** [comment]
Hi @juanvasquezreyes, have you found the cause of this issue? Or maybe an alternative solution?

We are affected by this issue in one of our environments and we also tried the oncePer condition without success. The weirdest part is the same configuration in other environments behaves as expected even without having to use the oncePer condition.

**Reply by NaurisSadovskis** [reply]
We found that this issue was related to number of replicas we ran. Moving back to a single replica fixed the issue, moving back broke it. However, our issue started relatively recently, so we suspected this was due to leader lease acquisition, so we did a rollout restart for the Argo Rollouts component and the problem went away. Maybe this helps someone.

**Reply by zachaller** [reply]
There was all sorts of issues with leader election that have been fixed in 1.4 there is an rc1 if you want to give it a test.

**Comment by blakebarnett** [comment]
We're seeing this with rollouts 1.5.1, it's consistent for all apps. We only have 1 argo-rollouts replica, and 1 argo-rollouts-dashboard replica. We have this in the notifications ConfigMap, I'm wondering if the abort is done automatically and we're causing the 2nd one my specifying it in here?

```
  trigger.on-analysis-run-error: |
    - send: [analysis-run-error]
  trigger.on-analysis-run-failed: |
    - send: [analysis-run-failed]
  trigger.on-analysis-run-running: |
    - send: [analysis-run-running]
  trigger.on-rollout-aborted: |
    - send: [rollout-aborted]
  trigger.on-rollout-completed: |
    - send: [rollout-completed]
  trigger.on-rollout-paused: |
    - send: [rollout-paused]
  trigger.on-rollout-step-completed: |
    - send: [rollout-step-completed]
  trigger.on-rollout-updated: |
    - send: [rollout-updated]
  trigger.on-scaling-replica-set: |
    - send: [scaling-replicaset]
```
